AEPCampaignClassic/Sources/CampaignClassic+PublicAPI.swift

```diff
@@ -16,4 +16,89 @@ import Foundation
 
 /// Defines the public interface for the CampaignClassic extension
 @objc public extension CampaignClassic {
+
+    /// Registers a device with the configured Adobe Campaign Classic server instance.
+    /// The `callback` will be executed when the registration request is completed, returning success status of the call.
+    ///
+    /// - Parameters:
+    ///    - token : A unique device token received after registering your app with APNs servers
+    ///    - userKey : A `string` containing the user identifier
+    ///    - additionalParameters : a dictionary of custom key-value pairs to be sent along with the registration call
+    ///    - callback : a block which will be called after the device registration is complete. The callback returns
```

Comment:
should we update the completion block to pass an optional `Error` in the case of failure?

Reply:
planned to remove the callback